File: cpp/hardware/cooling/LiquidCooling.cpp
#pragma once
#include <bits/stdc++.h>

#include "Cooling.cpp"

class LiquidCooling : public Cooling
{
private:
    double fluidVolume;
    std::string fluidType;
    bool hasPump;

public:
    // Constructor
    LiquidCooling() : Cooling(), fluidVolume(0.0), fluidType(""), hasPump(false) {}
    LiquidCooling(const std::string &serialNumber, const Date &installationDate, bool physicallyInstalled, int thermalDissipation, const std::string &id, const std::string &manufacturer, const std::string &model, double price, double volume, const std::string &type, bool pump)
        : Cooling(serialNumber, installationDate, physicallyInstalled, thermalDissipation, id, manufacturer, model, price), fluidVolume(volume), fluidType(type), hasPump(pump) {}

    // Destructor
    ~LiquidCooling() {}

    // Setters
    void setFluidVolume(double volume)
    {
        fluidVolume = volume;
    }

    void setFluidType(const std::string &type)
    {
        fluidType = type;
    }

    void setHasPump(bool pump)
    {
        hasPump = pump;
    }

    // Getters
    double getFluidVolume() const
    {
        return fluidVolume;
    }

    std::string getFluidType() const
    {
        return fluidType;
    }

    bool getHasPump() const
    {
        return hasPump;
    }
};

File: python/hardware/cooling/liquid_cooling.py
from datetime import date
from hardware.cooling.cooling import Cooling

class LiquidCooling(Cooling):
    def __init__(self, serial_number="", installation_date=None, physically_installed=False, max_thermal_dissipation=0, id="", manufacturer="", model="", price=0.0, fluid_volume=0.0, fluid_type="", has_pump=False):
        super().__init__(serial_number, installation_date, physically_installed, max_thermal_dissipation, id, manufacturer, model, price)
        self._fluid_volume = fluid_volume
        self._fluid_type = fluid_type
        self._has_pump = has_pump

    # Setters
    def set_fluid_volume(self, fluid_volume):
        self._fluid_volume = fluid_volume

    def set_fluid_type(self, fluid_type):
        self._fluid_type = fluid_type

    def set_has_pump(self, has_pump):
        self._has_pump = has_pump

    # Getters
    def get_fluid_volume(self):
        return self._fluid_volume

    def get_fluid_type(self):
        return self._fluid_type

    def get_has_pump(self):
        return self._has_pump